src/main/resources/org/opentripplanner/apis/transmodel/schema.graphql

```diff
@@ -2187,6 +2192,56 @@ input TripFilterSelectInput {
   transportModes: [TransportModes!]
 }
 
+"""
+One of the listed stop locations must be visited on-board a transit vehicle or the journey must
+alight or board at the location.
+"""
+input TripPassThroughViaLocationInput {
+  "The label/name of the location. This is pass-through information and is not used in routing."
+  label: String
+  """
+  A list of stop locations. A stop location can be a quay, a stop place, a multimodal
+  stop place or a group of stop places. It is enough to visit ONE of the locations
+  listed.
+  """
+  stopLocationIds: [String!]
+}
+
+"""
+A via-location is used to specifying a location as an intermediate place the router must
+route through. The via-location is either a pass-through-location or a visit-via-location.
+"""
+input TripViaLocationInput @oneOf {
+  "Board, alight or pass-through(on-board) at the stop location."
+  passThrough: TripPassThroughViaLocationInput
+  "Board or alight at the stop-locations/coordinate."
+  visit: TripVisitViaLocationInput
+}
```

Comment:
Two different spellings stop location and stop-locations are used here. Also, boarding or alighting a coordinate doesn't make sense.